modules/le_log/le_log.cpp

```diff
@@ -0,0 +1,94 @@
+#include "le_log.h"
+#include "le_core/le_core.h"
+
+#include <atomic>
+#include <mutex>
+#include <unordered_map>
+#include <cstdarg>
+
+struct le_log_module_o {
+	std::string     name      = "DEFAULT";
+	std::atomic_int log_level = 1;
+};
+
+static le_log_module_o *le_log_module_default() {
+	static auto module = new le_log_module_o();
+	return module;
+}
+
+static le_log_module_o *le_log_get_module( const char *name ) {
+	static std::unordered_map<std::string, le_log_module_o *> modules;
```

Comment:
i have a feeling that this will leak on hot-reload, as `modules` will be re-initialised, and all previously held elements in modules will get leaked, because the c++ delete operation will only delete the pointers to `le_log_module_o`, but not the objects themselves.

in any case, we don't want the objects to drop until the application is unloaded. the safest way to keep them around is to store a pointer to the collection with the api.

Reply:
Yes because these will be reinitialized, I was wondering about persistent static data. or maybe initialize one global pointer on startup. but since all other modules will probably rely on le log, this initialization method will always have to be called

Reply:
Re: persistent static data: [here](https://github.com/tgfrerer/island/blob/a9e7d2391b9c6e79d4a8a03dd2f1833b3776cdc1/modules/le_renderer/le_renderer.cpp#L730-L732
) is how I implemented that pattern for le_renderer's texture_handle_store.